src/psyclone/psyir/backend/fortran.py

```diff
@@ -192,6 +202,67 @@ def gen_declaration(self, symbol):
         result += "\n"
         return result
 
+    def container_node(self, node):
+        '''This method is called when a Container instance is found in
+        the PSyIR tree.
+
+        A container node is mapped to a module in the Fortran back end.
+
+        :param node: a Container PSyIR node.
+        :type node: :py:class:`psyclone.psyGen.Container`
+
+        :returns: the Fortran code as a string.
+        :rtype: str
+
+        :raises VisitorError: if the name attribute of the supplied \
+        node is empty or None.
+
+        '''
+        if not node.name:
+            raise VisitorError("Expected node name to have a value.")
+
+        # All children must be kernelschedule as modules within
+        # modules are not supported.
+        from psyclone.psyGen import KernelSchedule
+        if not all([isinstance(child, KernelSchedule)
+                    for child in node.children]):
+            raise VisitorError(
+                "The Fortran back-end requires all children of a Container "
+                "to be KernelSchedules.")
+
+        result = (
+            "{0}module {1}\n"
+            "".format(self._nindent, node.name))
+
+        self._depth += 1
+
+        declarations = ""
+        # Use statements
```

Comment:
Maybe expand to say these must come before variable declarations (otherwise a non-Fortran programmer may wonder why we don't just loop over the symbols once).

Reply:
Added a comment.